server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregationBuilder.java

```diff
@@ -144,6 +145,48 @@ public final BucketCardinality bucketCardinality() {
         }
     }
 
+    public abstract static class MetricsAggregationBuilder<VS extends ValuesSource, AB extends ValuesSourceAggregationBuilder<AB>> extends
+        LeafOnly<VS, AB> {
+
+        protected MetricsAggregationBuilder(String name) {
+            super(name);
+        }
+
+        protected MetricsAggregationBuilder(LeafOnly<VS, AB> clone, Builder factoriesBuilder, Map<String, Object> metadata) {
+            super(clone, factoriesBuilder, metadata);
+        }
+
+        protected MetricsAggregationBuilder(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        /** Generated metrics from this aggregation that can be accessed via
+         * {@link org.elasticsearch.search.aggregations.InternalAggregation#getProperty(String)}*/
```

Comment:
I got a little confused by this, because for bucket aggregations the story is more complex. But, like you say, for metric aggregations this is indeed the list.